src/util/__tests__/getMigratedMapConfig.spec.js

```diff
@@ -0,0 +1,175 @@
+            id: 'mapId',
+            name: 'map name',
+            basemap: { id: 'TheRainbowBasemap' },
+            mapViews: [
+                { layer: 'thematic', name: 'All the pretty colors' },
+                { layer: 'facilities', name: 'All the facilities' },
+            ],
+        })
+    );
+});
+
+test('getMigratedMapConfig when basemap is string "none"', () => {
+    const config = {
+        id: 'mapId',
+        name: 'map name',
+        basemap: 'none',
+        mapViews: [
+            { layer: 'thematic', name: 'All the pretty colors' },
+            { layer: 'facilities', name: 'All the facilities' },
+        ],
+    };
+
+    expect(getMigratedMapConfig(config, defaultBasemapId)).toEqual(
+        expect.objectContaining({
+            id: 'mapId',
+            name: 'map name',
+            basemap: {
+                id: defaultBasemapId,
+                isVisible: false,
+            },
+            mapViews: [
+                { layer: 'thematic', name: 'All the pretty colors' },
+                { layer: 'facilities', name: 'All the facilities' },
+            ],
+        })
+    );
+});
+
+test('getMigratedMapConfig when basemap is an object', () => {
+    const config = {
+        id: 'mapId',
+        name: 'map name',
+        basemap: {
+            id: 'osmStreet',
+            displayName: 'Basemap name',
+        },
+        mapViews: [
+            { layer: 'thematic', name: 'All the pretty colors' },
+            { layer: 'facilities', name: 'All the facilities' },
+        ],
+    };
+
+    expect(getMigratedMapConfig(config, defaultBasemapId)).toEqual(
+        expect.objectContaining({
+            id: 'mapId',
+            name: 'map name',
+            basemap: { id: 'osmStreet', displayName: 'Basemap name' },
+            mapViews: [
+                { layer: 'thematic', name: 'All the pretty colors' },
+                { layer: 'facilities', name: 'All the facilities' },
+            ],
+        })
+    );
+});
+
+test('getMigratedMapConfig when no basemap in config', () => {
+    const config = {
+        id: 'mapId',
+        name: 'map name',
+        mapViews: [
+            { layer: 'thematic', name: 'All the pretty colors' },
+            { layer: 'facilities', name: 'All the facilities' },
+        ],
+    };
+
+    expect(getMigratedMapConfig(config, defaultBasemapId)).toEqual(
+        expect.objectContaining({
+            id: 'mapId',
+            name: 'map name',
+            basemap: { id: defaultBasemapId },
+            mapViews: [
+                { layer: 'thematic', name: 'All the pretty colors' },
+                { layer: 'facilities', name: 'All the facilities' },
+            ],
+        })
+    );
+});
+
+// TODO - add a test that checks externalLayer and basemap
+test('getMigratedMapConfig with old GIS app format and Boundary layer', () => {
+    const config = {
+        id: 'mapId',
+        name: 'map name',
+        basemap: { id: 'osmStreet' },
```

Comment:
@turban Do the old configurations have a separate `basemap` property like this or is it in the mapViews (as in the first test in this file)

Reply:
What I know: 
- basemap prop should be a string (when new maps are saved)
- basemap prop should be "none" if the basemap visibility is off when saved 
- basemap prop is not always present (fallback to default basemap)
- basemap is google maps, then fallback to equivalent bing maps
- basemap is uid, use the external layer with the same uid
- basemap is object with id, only use the id (haven't seen this lately, could be used by the old GIS app)
- if basemap id is unkown (none of above), fallback to default basemap

For new maps, an external basemap is saved by storing the basemap uid in the basemap prop. External overlays are stored as a separate layer under mapViews. Previously also external basemaps were stored under mapViews, and there are still some saved maps using this format that we should still support. Try: "Malaria cases Sierra Leone terrain basemap".